### PR TITLE
Fresh pull request for sorting algorithms

### PR DESCRIPTION
**Pull request by Ahel2000**
# Description
Add insertion sort, selection sort and merge sort.

Issue #5 

## Type of change

Please delete options that are not relevant.

- [x] Added Algorithm
- [x] Folder Structure

# Added to SUMMARY.md (delete section if not needed)?

Have you added the page to SUMMARY.md if this is a new page and not a fix?

- [x] Yes
- [ ] No

### CODE REVIEW

**Comment by sadn1ck** [comment]
@Ahel2000 do we need the .java files? You already added code in the markdown files. @arnabsen1729 opinion?

File: src/SortingAlgorithms/InsertionSort/Insertion.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    public static boolean less(Comparable v , Comparable w){
+        return v.compareTo(w)<0 ;
+    }
+    
+    public static void exch(Comparable a[], int i, int j){
+        Comparable swap=a[i];
+        a[i]=a[j];
+        a[j]=swap;
+    }
+}
```
Comment (arnabsen1729) [inline src/SortingAlgorithms/InsertionSort/Insertion.java:34]: Remove this file.

File: src/SortingAlgorithms/InsertionSort/insertion.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
+# Insertion Sort
+
+## Example Problem
+
+Given an array ``arr = [a1,a2,a3,a4,a5...,an] `` , we need to sort the array the array with the help of Insertion Sort algorithm.
+
+---------------
```
Comment (arnabsen1729) [inline src/SortingAlgorithms/InsertionSort/insertion.md:7]: use <hr>

File: src/SortingAlgorithms/InsertionSort/insertion.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+At this position we stop since we have traversed through the entire array.
+
+
+-------------
+
+## Code for the Algorithm
+
+
+
+-----------
```
Comment (arnabsen1729) [inline src/SortingAlgorithms/InsertionSort/insertion.md:72]: here as well remove these with `<hr>`

File: src/SortingAlgorithms/MergeSort/Merge.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        
+        while(highPtr<=upperbound){
+            workplace[j++]=theArray[highPtr++];
+        }
+        
+        for(j=0;j<n;j++){
+            theArray[lowPtr+j]=workplace[j];
+        }
+    }
+}
```
Comment (arnabsen1729) [inline src/SortingAlgorithms/MergeSort/Merge.java:57]: File not needed

**Comment by Ahel2000** [comment]
I have made the changes @arnabsen1729

File: src/SortingAlgorithms/InsertionSort/insertion.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+//12 is greater than 10. Hence we stop.
+```
+At this position we stop since we have traversed through the entire array.
+
+
+<hr>
+
+## Code for the Algorithm
+<hr>
+### Code For Insertion Sort
```
Comment (sadn1ck) [inline src/SortingAlgorithms/InsertionSort/insertion.md:70]: Add some space here, markdown shenanigans

**Review by sadn1ck — CHANGES_REQUESTED** [review]
Check the comments, then I will merge. @arnabsen1729 have a look as well

**Review by sadn1ck — APPROVED** [review]
LGTM

**Comment by sadn1ck** [comment]
@allcontributors add @Ahel2000 for contributing to documentation

**Pull request merged**